Objects/abstract.c

```diff
@@ -733,6 +734,41 @@ int PyObject_CopyData(PyObject *dest, PyObject *src)
     return 0;
 }
 
+int
+PyObject_CopyToObject(PyObject *obj, void *buf, Py_ssize_t len,
+                      char Py_UNUSED(fort))
+{
+    Py_buffer view_obj;
+
+    if (!PyObject_CheckBuffer(obj)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "destination must be bytes-like objects");
+        return -1;
+    }
+
+    if (!buf) {
+        PyErr_SetString(PyExc_MemoryError, "buf pointer is null");
+        return -1;
+    }
+
+    if (PyObject_GetBuffer(obj, &view_obj, PyBUF_SIMPLE)) {
+        return -1;
+    }
+
+    if (view_obj.len < len) {
+        PyErr_SetString(PyExc_BufferError,
+                        "destination is too small to"
+                        "receive data from buf");
+        PyBuffer_Release(&view_obj);
+        return -1;
+    }
+
+    /* just copy it directly through memcpy */
+    memcpy(view_obj.buf, buf, len);
```

Comment:
You must consider the array orientation (both of the source and the destination):

> If fortran is 'F', then if the object is multi-dimensional, then the data will be copied into the array in Fortran-style (first dimension varies the fastest). 

> If fortran is 'C', then the data will be copied into the array in C-style (last dimension varies the fastest).

>  If fortran is 'A', then it does not matter and the copy will be made in whatever way is more efficient

We should probably reject other kind of ordering as well.

Reply:
I seem to have deleted my question. The important thing is that it seems that the fortran parameter is not very useful (only here). I only have one question, that is, does void have to be multi-dimensional or a continuous memory?

Reply:
Maybe you are right, Please give me some time. I'll made change.

Reply:
I have modified